src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/JavaScript/GethLikeBlockJavaScriptTracer.cs

```diff
@@ -68,8 +68,6 @@ public override void EndBlockTrace()
     {
         base.EndBlockTrace();
         Engine.CurrentEngine = null;
-        ArrayPoolList<IDisposable>? list = Interlocked.Exchange(ref _engines, null);
-        list?.Dispose();
```

Comment:
Ok you fixed test cases that didn't dispose `ArrayPoolList`, which is good. But now without this, don't we have potential issue when actually running the tracer in JSON RPC?

Reply:
I removed this as it looked like it would result in a double free if this is called, and then Dispose is called. Also it doesn't free each element of the list. So shouldn't the call in JSON RPC call Dispose?

Reply:
Also I think it doesn't just affects tests, I added some `TryDispose` to code outside of the tests